### PR TITLE
Rename Room#id -> Room#roomid, User#userid -> User#id

### PR DESCRIPTION
**Pull request by TheJetOU**
This rename makes the properties match their respective types.

### CODE REVIEW

**Comment by TheJetOU** [comment]
(Whoever merges and restarts the server must also rename them in private chat plugins.)

File: sim/dex-data.ts [hunk 1]
```diff
@@ -40,8 +40,8 @@ export class Tools {
 	static getId(text: any): ID {
 		if (text && text.id) {
 			text = text.id;
-		} else if (text && text.userid) {
```
Comment (whalemer) [inline sim/dex-data.ts]: I'm not sure if it matters anywhere but we might want to keep the .userid check

Comment (TheJetOU) [reply sim/dex-data.ts]: I guess it's still used for `muteQueue`, `BattleReady`, etc. Re-added.

**Comment by Zarel** [comment]
Okay let's do this tomorrow.

**Pull request merged**